test/javascripts/acceptance/chat-test.js

```diff
@@ -1515,18 +1515,32 @@ acceptance("Discourse Chat - image uploads", function (needs) {
     const appEvents = loggedInUser().appEvents;
     const done = assert.async();
 
-    appEvents.on("chat-composer:all-uploads-complete", () => {
+    appEvents.on("chat-composer:all-uploads-complete", async () => {
```

Comment:
Actually do you know if the test will block on this? I have a feeling that the test will move on after setting the listeners on the appEvents. That means by the time the assertions are called, it may not be the right screen anymore.